main.tf

```diff
@@ -69,6 +69,7 @@ data "template_file" "default" {
 }
 
 resource "aws_s3_bucket_policy" "default" {
+  count  = local.using_existing_origin && ! var.override_origin_bucket_policy ? 0 : 1
```

Comment:
@dmattia 
do we really need `var.override_origin_bucket_policy`?
If you specify `var.origin_bucket`, the policy will not be created as well as the bucket itself.
If you don't specify the origin bucket, we want to create it as well as the policy.

Reply:
I would prefer the approach you're talking about here, where we only manage the policy if we manage the bucket.

The reason I added the var is for backwards compatibility, as the current code in master _will_ manage the policy even if the bucket isn't managed.

If you're okay with removing var.override_origin_bucket_policy, I'll gladly do it!

@aknysh

Reply:
```
count  = local.using_existing_origin && ! var.override_origin_bucket_policy ? 0 : 1
```

In the expression above, if the existing bucket is provided, the policy will not be created because `local.using_existing_origin` is `true`, so `! var.override_origin_bucket_policy` is completely redundant 

Please update the logic.
So yes, we want to manage the policy if we create the bucket.
And make it configurable to manage the policy or not when an existing bucket is provided.
(this is not the expression above does. It will never create the policy if an existing bucket is provided).



Thanks

Reply:
(Apologies for the delay, I was OOO).

Thanks for the correction! I updated the logic, not sure how I made that mistake the first time round